draft-ietf-tls-esni.md

```diff
@@ -140,41 +141,43 @@ Client <------------------------------------>|                     |
 ~~~~
 {: #split-mode title="Split Mode Topology"}
 
-In Split Mode, the provider is *not* the origin server for private domains.
-Rather the DNS records for private domains point to the provider, and the
-provider's server relays the connection back to the backend server, which is the
-true origin server. The provider does not have access to the plaintext of the
-connection.
+In Split Mode, the provider is not the origin server for private domains.
+Rather, the DNS records for private domains point to the provider, and the
+provider's server relays the connection back to the origin server, who
+terminates the TLS connection with the client.
+
+In the remainder of this document, we will refer to the ECH-service provider as
+the "client-facing server" and to the TLS terminator as the "backend server".
+These are the same entity in Shared Mode, but in Split Mode, the client-facing
+and backend servers are physically separated. However, we will require them to
+communicate over a long-lived secure channel. (The manner in which this channel
+is established is out of the scope of this document.)
```

Comment:
When I first read this document, I thought the fact that the SNI was sent in the clear from the client-facing server to the backend server was an attack. As such, I think we want to be explicit about this assumption. This seems like a reasonable place to do so since we're talking about the parties in the protocol and how they communicate with one another.

Reply:
There's a couple reasons I push back here: (1) the connection might not be long-lived, and (2) the threat we're mostly concerned with is the client<->client-facing-server snooper (an adversary which sees client-facing-server<->backend connections might learn the SNI if it were not encrypted, but it wouldn't learn the client's IP. Basically, this is too restrictive of a threat model, and somewhat orthogonal to the protocol itself.